social_core/utils.py

```diff
@@ -235,10 +232,9 @@ def setting_url(backend, *names):
     for name in names:
         if is_url(name):
             return name
-        else:
-            value = backend.setting(name)
-            if is_url(value):
-                return value
+        value = backend.setting(name)
+        if is_url(value):
+            return value
```

Comment:
I'd add explicit return here:

```suggestion
            return value
        return None
```